addon/authenticators/oauth2-password-grant.js

```diff
@@ -325,7 +325,13 @@ export default BaseAuthenticator.extend({
     return new RSVP.Promise((resolve, reject) => {
       fetch(url, options).then((response) => {
         response.text().then((text) => {
-          let json = text ? JSON.parse(text) : {};
+          var json;
+          try {
+            json = JSON.parse(text);
+          } catch (e) {
```

Comment:
I've been battling this issue as well, so this is a good change, but this too broad. Catching all exceptions means that _any_ failure will be swallowed. Only a `SyntaxError` should be caught.

Reply:
good point 👍